mlrun/utils/notifications/notification/webhook.py

```diff
@@ -63,6 +63,33 @@ async def push(
             request_body["custom_html"] = custom_html
 
         if override_body:
+            list_edit_runs = []
+            for run in runs:
+                if isinstance(run, mlrun.model.RunObject):
+                    run = run.to_dict()
+                if isinstance(run, dict):
+                    r = {
+                        "project": run["metadata"]["project"],
+                        "name": run["metadata"]["name"],
+                        "host": run["metadata"]["labels"]["host"],
+                        "status": {"state": run["status"]["state"]},
+                    }
+                    if run["status"].get("error", None):
+                        r["status"]["error"] = run["status"]["error"]
+                    elif run["status"].get("results", None):
+                        r["status"]["results"] = run["status"]["results"]
+                    run = r
+                list_edit_runs.append(run)
```

Comment:
Not sure why you are appending a run if its not a dict but here is a suggestion to improve this block
```suggestion
                    parsed_run = {
                        "project": run["metadata"]["project"],
                        "name": run["metadata"]["name"],
                        "host": run["metadata"]["labels"]["host"],
                        "status": {"state": run["status"]["state"]},
                    }
                    if run["status"].get("error", None):
                        parsed_run["status"]["error"] = run["status"]["error"]
                    elif run["status"].get("results", None):
                        parsed_run["status"]["results"] = run["status"]["results"]
                    list_edit_runs.append(parsed_run)
```